sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildInstaller.java

```diff
@@ -145,67 +149,45 @@ private File locateZip(Version scannerVersion, boolean useDotNetCore) {
       }
     } else {
       LoggerFactory.getLogger(ScannerForMSBuildInstaller.class).info("Searching for Scanner for MSBuild {} in maven repository", scannerVersion);
-      zipFile = locators.locate(mavenLocation(scannerVersion, useDotNetCore));
+      zipFile = locators.locate(mavenLocation(scannerVersion, packageDetails));
     }
     return zipFile;
   }
 
-  static MavenLocation mavenLocation(Version scannerVersion) {
-    return mavenLocation(scannerVersion, false);
-  }
-
-  static MavenLocation mavenLocation(Version scannerVersion, boolean useDotNetCore) {
-    String groupId = "org.sonarsource.scanner.msbuild";
-    String artifactId = "sonar-scanner-msbuild";
+  static MavenLocation mavenLocation(Version scannerVersion, PackageDetails packageDetails) {
     MavenLocation.Builder location = MavenLocation.builder()
-      .setGroupId(groupId)
-      .setArtifactId(artifactId)
+      .setGroupId(packageDetails.getGroupId())
+      .setArtifactId(packageDetails.getArtifactId())
       .setVersion(scannerVersion.toString())
       .withPackaging("zip");
     if (scannerVersion.isGreaterThanOrEquals(DOT_NET_CORE_INTRODUCTION_MAJOR_VERSION, DOT_NET_CORE_INTRODUCTION_MINOR_VERSION)) {
-      location.setClassifier(getClassifier(useDotNetCore));
+      location.setClassifier(packageDetails.getClassifier());
     }
     return location.build();
   }
 
-  private static String getClassifier(boolean useDotNetCore) {
-    return useDotNetCore ? NETCOREAPP_2_0 : NET_46;
-  }
-
-  private static void clearCachedSnapshot(@Nullable Version scannerVersion, File toDir, boolean useDotNetCore) {
-    File scannerDir = new File(toDir, directoryName(scannerVersion, useDotNetCore));
-    if ((scannerVersion == null || scannerVersion.isSnapshot()) && scannerDir.exists()) {
+  private static void clearCachedSnapshot(@Nonnull File toDir, @Nonnull String scannerFolderName, boolean shouldDelete) {
```

Comment:
I wanted to reply "to avoid duplication" and then I realized that the usage is actually different. In the first call we always want to delete the folder since we don't know what's inside and on the second call we want to delete the folder only for snapshots.

So yeah, I've removed the parameter 😄